src/magic_folder/config.py

```diff
@@ -131,7 +133,8 @@ class SnapshotNotFound(Exception):
 ## sure how to do that w/o docs here
 
 
-def create_global_configuration(basedir, api_endpoint, tahoe_node_directory):
+def create_global_configuration(basedir, api_endpoint, tahoe_node_directory,
+                                api_client_endpoint=None):
```

Comment:
I'd consider making this a required argument and doing the conversion in the option parsing code.  That way if automatic conversion fails and no value is supplied, it's the option parsing code that can fail - reporting a usage error.  And this code is safe and free of the extra conditionals for handling bogus input/combination of inputs.

Put another way (more formally or maybe even objectively), the option parsing code is already partial (it will fail in some inputs) but this function is currently total (except for I/O failures).  Moving the checks on this value into the option parsing code leaves that code partial and leaves this function total instead of making them both partial as they are here.

Also, it looks like this is a string description and not an endpoint itself so a different name might be preferable.  The same goes for `api_endpoint`.